ToDo.js

```diff
@@ -0,0 +1,376 @@
+    modalHTML;
+
+    constructor() {
+        this.modalContainer = document.querySelector(".modal-container")
+        this.modalHTML = ` <div class="modal fade" id="cardModal" tabindex="-1" aria-labelledby="cardModalLabel" aria-hidden="true">
+        <div class="modal-dialog">
+          <div class="modal-content">
+            <form class="card-form">
+              <div class="form-group py-3">
+                <label for="taskTitle">Title</label>
+                <input type="text" class="form-control card-title" id="taskTitle" placeholder="Task title">
+              </div>
+              <div class="form-group">
+                <label for="taskDescription">Description</label>
+                <input type="text" class="form-control card-description" id="taskDescription"
+                  placeholder="Task description">
+              </div>
+            </form>
+            <div class="card-modal__footer d-flex justify-content-end">
+              <button type="button" class="btn modal__btn-delete position-absolute" data-bs-dismiss="modal">delete</button>
+              <button type="button" class="btn modal__btn-cancel card-modal__btn" data-bs-dismiss="modal">cancel</button>
+              <button type="button" class="btn modal__btn-save card-modal__btn" data-bs-dismiss="modal">save</button>
+            </div>
+          </div>
+        </div>
+       </div>`
+    }
+
+    generateModal() {
+        this.modalContainer.innerHTML = ""
+        this.modalContainer.insertAdjacentHTML('beforeend', `${this.modalHTML}`)
+    }
+}
+
+class ColumnsRendering {
+    columnsContainer;
+    constructor() {
+        this.columnsContainer = document.getElementById("columnContainer")
+    }
+
+    generateColumnHTML(colIndex, colName, colID) {
+        this.columnsContainer.insertAdjacentHTML('beforeend', `<div id="col_${colIndex}" data-column="${colID}" class="col-3 m-3 column rounded overflow-hidden position-relative">
+    <input type="text" value="${colName}" class="column__header column_content d-flex justify-content-center" placeholder="Column">
+      <button class="btn column_content column__btn p-0 m-0 d-flex align-items-center justify-content-center rounded-circle position-absolute"
+      data-bs-toggle="modal" data-bs-target="#cardModal"><img class=""
+              src="./Sources/CUSTOM_plusSign.png">
+      </button>
+      </button>
+      <button class="btn column_content column__delete-btn p-0 m-0 d-flex align-items-center justify-content-center position-absolute">
+          delete
+      </button>
+  </div>`)
+    }
+
+    async generateColumns(requestsObject) {
+        this.columnsContainer.innerHTML = ""
+        const columnsObj = await requestsObject.getColumnsObj()
+        columnsObj.map((column, index) => {
+            this.generateColumnHTML(index, column.name, column.uuid);
+        })
+    }
+
+    returnColumns() {
+        const columnsList = document.querySelectorAll(".column")
+        return columnsList
+    }
+}
```

Comment:
Bad naming convention, just UIColumn, UICard will be enough

Reply:
You could separate logic into - mvc

View would handle UI related logic
Controller would pass data from model to view and handle listeners
Model that is data layer.

Reply:
If you have issues understanding difference between view logic and controller logic:

- view doesn't care about business logic, it just displays data that match contract
- controller takes business data and process it so it's consumable by view